### ISSUE TITLE
[coap] use of `nullptr` in C API

### ISSUE THREAD

**Issue by krbvroc1** [openthread/openthread#6171]
I just fetched the most recent Zephyr zephyr-v2.5.0-148-g4ee876b797b1 which just merged #4247 (CoAP block transfers). 

Build fails... (board is nrf52)...both crosstools arm compiler and arm gcc.
```
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h: In function 'otCoapSendRequestBlockWise':
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h:946:108: error: 'nullptr' undeclared (first use in this function)
  946 |     return otCoapSendRequestBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, aHandler, aContext, nullptr,
      |                                                                                                            ^~~~~~~
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h:946:108: note: each undeclared identifier is reported only once for each function it appears in
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h: In function 'otCoapSendResponseBlockWise':
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h:1107:89: error: 'nullptr' undeclared (first use in this function)
 1107 |     return otCoapSendResponseBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, nullptr, aContext,
      |                                                                                         ^~~~~~~
```

**Comment by jwhui** [comment]
@krbvroc1 , thanks for raising this issue. I just submitted a proposed fix in #6173, please review.

**Comment by krbvroc1** [comment]
Thanks.
